### PR TITLE
RM-89097 RM-89096 Release over_react 4.0.0

### PR DESCRIPTION
**Pull request by rmconsole-readonly-wk**
This stable, __major__ release of over_react includes:

### ReactJS 17.x Support

The underlying `.js` files provided by the react-dart 6.0.0 release are now ReactJS version `17.0.1`.

---

### Breaking Changes

React 17 includes some breaking changes, but rather than being large API removals, they are mostly subtle behavior changes of existing APIs. Please refer to their [blog post](https://reactjs.org/blog/2020/08/10/react-v17-rc.html#other-breaking-changes) for all behavior changes.

#### __Dart API Breaking Changes__

All of the Dart API breaking changes come from re-exports of members of the `react` package. Check out the [react 6.0.0 breaking changes](https://github.com/cleandart/react-dart/pull/285) for more information.

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Comment by greglittlefield-wf** [comment]
### Public API Changes

<p><em><strong>Release PR detected; using previous release (3.12.1) as the diff base.</strong></em></p> ## Issue: No Semver Audit Report Found for Base Commit

**Base commit:** `4e97b41568645ff9edaf117a9e4e04f127b18f5d`

This means we couldn't find semver audit data for the commit against which to compare the changes in this PR. Fixing this is a two step process. First, we'll need to generate the semver audit data for the base commit. Then we will have to generate the semver audit data for the head commit on this branch. This can be done by rerunning CI and/or manually running the semver audit tool.

**Note: It may be sufficient to simply merge master into your branch.**

## Step 1: Generate Semver Audit Data for the Base Commit:

You can do this either by rerunning CI on the base commit or by manually running the semver audit tool locally on the base commit.

### Option 1: Rerunning CI

Locate the build associated with the base commit (4e97b41568645ff9edaf117a9e4e04f127b18f5d) in [Workiva Build](https://ci.webfilings.com/Workiva/over_react) and rerun it.

### Option 2: Run the Semver Audit Tool Manually

```bash
# Navigate to the root of the repo and check out the base commit:
$ git checkout 4e97b41568645ff9edaf117a9e4e04f127b18f5d

# If this is a Dart project, run:
$ pub global run semver_audit report --repo Workiva/over_react --head-commit 4e97b41568645ff9edaf117a9e4e04f127b18f5d

# If this is a Go project, run:
$ semver_audit_go report  --repo Workiva/over_react --head-commit 4e97b41568645ff9edaf117a9e4e04f127b18f5d
```

**Note:** If this is a Dart project, the manual steps above assume you have the Dart 2 SDK installed and activated and the `semver_audit` tool globally installed. See [this Confluence post](https://wiki.atl.workiva.net/questions/104006659/how-do-i-install-dart-2-and-easily-switch-between-dart-versions) to install Dart 2. To globally install the `semver_audit` tool, run the following:

```bash
$ pub global activate --hosted-url https://pub.workiva.org semver_audit ^2.0.0
```

## Step 2: Generate Semver Audit Data for this Branch's Head Commit:

You can do this either by rerunning CI on this PR or by manually running the semver audit tool locally against the head commit on this branch.

### Option 1: Rerunning CI

This is often the easiest solution. Find the Workiva Build check on this Pull Request and simply rerun it.

### Option 2: Run the Semver Audit Tool Manually

```bash
# Navigate to the root of the repo and check out the head commit on this branch:
$ git checkout 7c55759ac7ff837587060be20829525bcc238f64

# If this is a Dart project, run:
$ pub global run semver_audit report --repo Workiva/over_react --base-commit 4e97b41568645ff9edaf117a9e4e04f127b18f5d --head-commit 7c55759ac7ff837587060be20829525bcc238f64 --pull 647

# If this is a Go project, run:
$ semver_audit_go report --repo Workiva/over_react --base-commit 4e97b41568645ff9edaf117a9e4e04f127b18f5d --head-commit 7c55759ac7ff837587060be20829525bcc238f64 --pull 647
```

## Still Need Help?

If you are unable to resolve the issue on your own, either tag `@Workiva/app-frameworks` in a PR comment or reach out to us in the [#support-client-plat](https://slack.com/app_redirect?channel=CEDM9RH1N) Slack channel.


Showing results for 7c55759ac7ff837587060be20829525bcc238f64
> Powered by [semver-audit-service](https://github.com/Workiva/semver-audit-service). Please report any problems by [filing an issue](https://github.com/Workiva/semver-audit-service/issues).
> Reported by the dart semver audit client 2.2.0
[Browse public API](https://semver-audit.wk-dev.wdesk.org/Workiva/over_react/7c55759ac7ff837587060be20829525bcc238f64).

**Comment by greglittlefield-wf** [comment]
### Public API Changes

 Recommendation: **:bangbang: Major version bump (fyi @Workiva/semver-audit-group )**


```diff
@@ line 24: file:///Users/greglittlefield/workspaces/over_react4/lib/src/util/event_helpers.dart @@
-  dynamic wrapNativeKeyboardEvent(KeyboardEvent nativeKeyboardEvent)
// Removing a top-level function is a major change.
```

```diff
@@ line 79: file:///Users/greglittlefield/workspaces/over_react4/lib/src/util/event_helpers.dart @@
-  dynamic fakeSyntheticFormEvent(Element element, String type)
// Removing a top-level function is a major change.
```

```diff
@@ line 52: file:///Users/greglittlefield/workspaces/over_react4/lib/src/util/event_helpers.dart @@
-  dynamic wrapNativeMouseEvent(MouseEvent nativeMouseEvent)
// Removing a top-level function is a major change.
```

Showing results for 7c55759ac7ff837587060be20829525bcc238f64
> Powered by [semver-audit-service](https://github.com/Workiva/semver-audit-service). Please report any problems by [filing an issue](https://github.com/Workiva/semver-audit-service/issues).
> Reported by the dart semver audit client 2.2.0
[Browse public API](https://semver-audit.wk-dev.wdesk.org/Workiva/over_react/7c55759ac7ff837587060be20829525bcc238f64).

**Comment by joebingham-wk** [comment]
All commits after 03d73c8 are unreviewed - but the only changes are to the changelog, the root pubspec, and the todo app pubspec

**Review by sydneyjodon-wk — APPROVED** [review]
+1

File: lib/src/util/hoc.dart [hunk 1]
```diff
@@ -45,6 +45,6 @@
 ///   ),
 /// ])(_$BigBlock);
 /// ```
-R Function(A) composeHocs<R, A extends R>(Iterable<R Function(A)> functions) {
+R Function(R) composeHocs<R>(Iterable<R Function(R)> functions) {
```
Comment (greglittlefield-wf) [inline lib/src/util/hoc.dart:48]: This change fixes a bad implicit cast in the implementation and shouldn't affect consumers.

Thankfully, there are no consumers, so we know it won't affect any 😄 : https://sourcegraph.wk-dev.wdesk.org/search?q=composeHocs&patternType=literal

**Review by aaronlademann-wf — APPROVED** [review]
+1

**Comment by aaronlademann-wf** [comment]
Semver +1

**Review by greglittlefield-wf — APPROVED** [review]
+10

@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**